templates/index.tmpl

```diff
@@ -0,0 +1,7 @@
+{{ define "home" }}
+{{ template "header" . }}
```

Comment:
What's the `.` ? just path of file for the defined variable? Is that needed if you already defined it in `template.ParseFiles`?

Reply:
`.` Refers to the object being passed into the template.
At the caller I'm passing in the struct, but within the template itself, the struct passed in is referred to as `.`.
Any partials being rendered by the invoked template don't receive the initial object passed in, so this is how you accomplish that.

I'm kind of stumbling over my own words there.
If you look at the first example shown in the overview here: https://golang.org/pkg/text/template/
See how the attribute is accessed via `.Material`?  We don't have a name for the object being passed in... I think this way of just invoking the attribute on nothing (ie., `.Material`) or referring to the currently active object via `.` is just special syntax exclusive to templates in Golang.  I'm honestly not too sure.